### DISCUSSION TITLE
Request for Docker example with Sharp

### DISCUSSION THREAD

**Discussion by vasanthi-d** [vercel/next.js#40126] [Help]
Nextjs docs do not mention much about Sharp error. There is only one link https://nextjs.org/docs/messages/sharp-missing-in-production that talks about installing sharp and adding environment variable.

I have found many posts where many have faced issues in getting it work. After almost spending 2 days on it, I still couldn't get it work.
I tried both '`next start`' and `standalone` modes with Next/Image fr images.

Hence, requesting for a  Docker example with Sharp that would help many people who want to use Next/Image but have issues in getting it work?

Also, is 'Sharp' is required for Next/Image, why can't it be included as part of Next.js? why do we have explicitly '`npm install sharp`' and pass environment variable?

**Comment by icyJoseph** [comment]
The last one is a good question, I've never seen a very satisfactory answer, other than `squoosh` being kind of slow, but friendlier in a development environment, because it is a pure JS library (AFAIK), whereas `sharp` is just plain fast, no questions asked, based on C++ modules.

The odd thing is that all you should need to do is install sharp as a dependency and then you are good to go. We have a couple of projects with next/image, and sharp installed, with no special consideration in the Docker files what so ever. The env variable is only needed if you have sharp installed through some other means, perhaps you build it from source and symlink to it in your system, then you could just tell next.js where it is.

I'll try to investigate a bit more into this later today.. So far, I can't trigger the error. If you got a repository to share, or could make a minimal example, then we can get to the bottom of this much faster.

**Reply by vasanthi-d** [reply]
@icyJoseph - here is the question https://github.com/vercel/next.js/discussions/40116 I posted with my specific code snippets and setup

**Comment by bonesoul** [comment]
i'm also looking for this, @vasanthi-d were you able to fix this up?

**Reply by vasanthi-d** [reply]
@bonesoul No. After spending significant amount of time on this, I gave up and just used `standalone` mode to start the app without `sharp`. I am definitely interested in solution for this and would love to use `'next start'` with `sharp` if there is any performance gain.

**Reply by bonesoul** [reply]
i guess i found the solution. 
if you are targeting alpine in your builds, after npm install use this:

```
  npm install && \
  npm rebuild --platform=linux --arch=x64 --libc=musl sharp && \
```
alpine linux needs musl for sharp.
